sphinx_needs/directives/need.py

```diff
@@ -356,6 +359,16 @@ def previous_sibling(node: nodes.Node) -> Optional[nodes.Node]:
     return node.parent[i - 1] if i > 0 else None  # type: ignore
 
 
+def need_node_excluded_by_only_directive(env: BuildEnvironment, node: nodes.Node) -> bool:
+    """Return True if the node is under an "only" directive that shall be excluded given to the current tags"""
+    if hasattr(node.parent, "tagname") and node.parent.tagname == "only":
```

Comment:
I guess this would not support nested needs or needs in other directives like `container`, `list-table` and co.